core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.security.authentication;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.exception.ExceptionMessage;
+import alluxio.util.CommonUtils;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.Sets;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+
+import javax.annotation.concurrent.ThreadSafe;
+import javax.security.sasl.AuthenticationException;
+
+/**
+ * An authenticator for impersonation users.
```

Comment:
can you add more javadoc for the details of this authenticator?
e.g., when the authenticate will work and when will it will fail?